### ISSUE TITLE
Move token-action from /token to api/v1.0/token.

### ISSUE THREAD

**Issue by SvenHarenburg** [SvenHarenburg/SimpleKB.API#8]
As all other actions the token-generation should be moved to the api/{version}/-url.

**Comment by SvenHarenburg** [comment]
To move the action from /token to /api/V1.0/token I could simply change TokenProviderOptions.Path from "/token" to "/api/V1.0/token". 

This however might require additional work if I will ever introduce another version of the api. I will think about this and find a way that will make it easier for future me to add new versions.

**Comment by SvenHarenburg** [comment]
- Added a new Enum called "APIVersion" containing all supported api-versions. Due to language-restrictions I had to name them after the following format:
    - APIVersion.V10 = Version 1.0
    - APIVersion.V11 = Version 1.1
I also added an extension for that Enum which returns the description set through the DescriptionAttribute on the enum-members. This way I can easily define strings that can be used for example to build the token url.

-Introduced 2 new variables to TokenProviderOptions:
    -UseVersioning: Defines if the path will be build dynamically using the Version set in the Version property. If not the path will just return "/api/token".
    -Version: If UseVersioning is set to true, the value of this property will be used to dynamically create the path. Format: _"/api/{Version.GetDescription()}/token"_

If I will ever need to use different logic for the token-generation based on the version, I would have to simply check the version set in the TokenProviderOptions-parameter given to the TokenProviderMiddleware. Accordingly to the value I could then call different logic.
